taskcluster_util/util/downloader.py

```diff
@@ -54,23 +54,35 @@ def download_latest_artifact(self, task_id, full_filename, dest_dir):
         sys.stdout.flush()
 
         # move file to dest folder
-        abs_dest_dir = os.path.abspath(dest_dir)
-        log.debug('dest dir: {}'.format(abs_dest_dir))
-        if os.path.exists(abs_dest_dir) and (not os.path.isdir(abs_dest_dir)):
-            log.warning('Not a directory: {}'.format(abs_dest_dir))
-            final_file_path = os.path.abspath(temp_local_file)
-            log.debug('local file: {}'.format(final_file_path))
-        else:
-            if not os.path.exists(abs_dest_dir):
-                os.makedirs(abs_dest_dir)
-            shutil.copy(temp_local_file, abs_dest_dir)
-            final_file_path = os.path.join(abs_dest_dir, base_filename)
-            log.debug('local file: {}'.format(final_file_path))
-
-        # remove temp folder
+        final_file_path = temp_local_file
+        abs_dest_dir = os.path.abspath(dest_dir) if dest_dir else os.getcwd()
         try:
-            shutil.rmtree(self.temp_dir)  # delete directory
-        except OSError:
-            log.warning('Can not remove temporary folder: {}'.format(self.temp_dir))
+            FolderHandler(abs_dest_dir).copy_elements_from(temp_local_file)
+            final_file_path = os.path.join(abs_dest_dir, base_filename)
+            # remove temp folder
+            try:
+                shutil.rmtree(self.temp_dir)  # delete directory
```

Comment:
By the way, I still don't understand why we copy and then delete the folder. `shutil.move` already [handles that for us](https://docs.python.org/2.7/library/shutil.html?highlight=shutil#shutil.move).